lib/SARIF/Result.h

```diff
@@ -0,0 +1,58 @@
+//
+// Created by Daniel Dominguez on 11/1/23.
+//
+
+#ifndef VANGUARD_RESULT_H
+#define VANGUARD_RESULT_H
+
+#include "SARIF.h"
+#include "Object.h"
+#include "Message.h"
+#include "Location.h"
+#include <string>
+#include <utility>
+#include <vector>
+#include <memory>
+
+namespace vanguard {
+    class SARIF::Result : public SARIF::Object {
+    public:
+        using Message = SARIF::Message;
+        using Location = SARIF::Location;
+
+        Result(std::string detectorName, SARIF::Level severityLevel, std::unique_ptr<Message> message,
+                    std::vector<std::unique_ptr<Location>> locations) :
+                detectorName(std::move(detectorName)), severityLevel(severityLevel), message(std::move(message)),
+                locations(std::move(locations)) {};
+        Result(std::string detectorName, SARIF::Level severityLevel, std::unique_ptr<Message> message,
+               std::unique_ptr<Location> location) :
+                detectorName(std::move(detectorName)), severityLevel(severityLevel), message(std::move(message)) {
+            locations.push_back(std::move(location));
+        };
+        Result(SARIF::Result &other) :
+                detectorName(other.detectorName),
+                severityLevel(other.severityLevel),
+                message(std::move(other.message)),
+                locations(std::move(other.locations)) {};
+
+        std::string getDetectorName() { return detectorName; }
+        std::string getSeverityLevel() {
+            switch(severityLevel) {
+                case ERROR: return "error";
+                case WARNING: return "warning";
+                case NOTE: return "note";
+                case NONE: return "none";
+            }
+        }
+        std::unique_ptr<Message> getMessage() { return std::move(message); }
+        std::vector<std::unique_ptr<Location>> getLocations() { return std::move(locations); }
```

Comment:
I don't think these should move the variables, otherwise there's no way to "read" the result without modifying it.
```suggestion
        const Message *getMessage() const { return message.get(); }
        const std::vector<std::unique_ptr<Location>> &getLocations() const { return locations; }
```

If we really need some sort of move method, it would be better to rename (or copy) the current implementation as `takeMessage/Locations()`.